stellar/txobserver/main.go

```diff
@@ -5,47 +5,55 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/pkg/errors"
 	"github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/protocols/horizon"
 	slog "github.com/stellar/go/support/log"
 	"github.com/stellar/starbridge/store"
 )
 
 type Observer struct {
 	ctx context.Context
 
+	bridgeAccount string
+
 	client *horizonclient.Client
 	store  *store.DB
 	log    *slog.Entry
 
-	// TODO: this will be persisted in a DB
 	ledgerSequence uint32
 }
 
-func NewObserver(ctx context.Context, client *horizonclient.Client, store *store.DB) *Observer {
+func NewObserver(ctx context.Context, bridgeAccount string, client *horizonclient.Client, store *store.DB) *Observer {
 	o := &Observer{
-		ctx:    ctx,
-		client: client,
-		store:  store,
-		log:    slog.DefaultLogger.WithField("service", "stellar_txobserver"),
+		ctx:           ctx,
+		bridgeAccount: bridgeAccount,
+		client:        client,
+		store:         store,
+		log:           slog.DefaultLogger.WithField("service", "stellar_txobserver"),
 	}
 
-	root, err := o.client.Root()
+	ledgerSeq, err := o.store.GetLastLedgerSequence(context.Background())
 	if err != nil {
-		o.log.Fatalf("Unable to access Horizon (%s) root resource: %v", client.HorizonURL, err)
+		o.log.Fatalf("Unable to load last ledger sequence from db: %v", err)
 	}
 
-	o.ledgerSequence = uint32(root.HorizonSequence)
+	if ledgerSeq == 0 {
```

Comment:
if the db is empty I think we should ingest from the ledger sequence where the bridge account was created. Maybe we can add a `bridgeAccountBirthSequence` parameter to `NewObserver()` ?